src/Workspaces/Core/Portable/Workspace/Solution/ProjectState.cs

```diff
@@ -547,9 +547,12 @@ public ProjectState WithParseOptions(ParseOptions options)
                 return this;
             }
 
+            var onlyPreprocessorDirectiveChange = ParseOptions != null &&
+                (_languageServices.SyntaxTreeFactory?.OptionsDifferOnlyByPreprocessorDirectives(options, ParseOptions) ?? false);
```

Comment:
if we have parseOptions can SyntaxTreeFactory be null? :(

Reply:
Ah, hopefully not; will switch to !.